### PR TITLE
Moved detect_providers to providers.py

### PR DESCRIPTION
**Pull request by tonybaltovski**
This allows the job to also detect the init system.

### CODE REVIEW

**Review by paulbovbel — APPROVED** [review]

File: src/robot_upstart/job.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def add(self, package=None, filename=None, glob=None):
             for path in search_paths:
                 self.files.extend(glob_files(os.path.join(path, glob)))
 
-    def install(self, root="/", sudo="/usr/bin/sudo", Provider=providers.Upstart):
+    def install(self, root="/", sudo="/usr/bin/sudo", Provider=providers.detect_provider()):
```
Comment (paulbovbel) [inline src/robot_upstart/job.py]: good practice in python not to call functions for default args, but specify None instead and do

```
if not Provider:
  Provider = detect_provider()
```

See http://www.informit.com/articles/article.aspx?p=2314818 item 20

Comment (mikepurvis) [reply src/robot_upstart/job.py]: +1

Otherwise looks good.

File: src/robot_upstart/providers.py [hunk 2]
```diff
@@ -34,6 +34,13 @@
 
 from catkin.find_in_workspaces import find_in_workspaces
 
+def detect_provider():
+    cmd=open('/proc/1/cmdline', 'rb').read().split('\x00')[0]
```
Comment (mikepurvis) [inline src/robot_upstart/providers.py:38]: Am surprised this passed roslint. Either way, please fix.

Comment (mikepurvis) [reply src/robot_upstart/providers.py:38]: Oh, there are a bunch of linter problems right now.

Nevermind. I will merge this and fix them in a batch. Thanks!

Comment (tonybaltovski) [reply src/robot_upstart/providers.py:38]: I should of ran the test before I committed, thanks for catching it. Applied fixed in #47.  Maybe we should enable Travis for this repo?

**Pull request merged**